pandas/tests/series/indexing/test_setitem.py

```diff
@@ -536,22 +557,30 @@ def test_setitem_keep_precision(self, any_numeric_ea_dtype):
         expected = Series([1, 2, 10], dtype=any_numeric_ea_dtype)
         tm.assert_series_equal(ser, expected)
 
-    @pytest.mark.parametrize("indexer", [1, 2])
     @pytest.mark.parametrize(
-        "na, target_na, dtype, target_dtype",
+        "na, target_na, dtype, target_dtype, indexer, warn",
         [
-            (NA, NA, "Int64", "Int64"),
-            (NA, np.nan, "int64", "float64"),
-            (NaT, NaT, "int64", "object"),
-            (np.nan, NA, "Int64", "Int64"),
-            (np.nan, NA, "Float64", "Float64"),
-            (np.nan, np.nan, "int64", "float64"),
+            (NA, NA, "Int64", "Int64", 1, None),
+            (NA, NA, "Int64", "Int64", 2, None),
+            (NA, np.nan, "int64", "float64", 1, FutureWarning),
+            (NA, np.nan, "int64", "float64", 2, None),
+            (NaT, NaT, "int64", "object", 1, FutureWarning),
+            (NaT, NaT, "int64", "object", 2, None),
+            (np.nan, NA, "Int64", "Int64", 1, None),
+            (np.nan, NA, "Int64", "Int64", 2, None),
+            (np.nan, NA, "Float64", "Float64", 1, None),
+            (np.nan, NA, "Float64", "Float64", 2, None),
+            (np.nan, np.nan, "int64", "float64", 1, FutureWarning),
+            (np.nan, np.nan, "int64", "float64", 2, None),
         ],
     )
-    def test_setitem_enlarge_with_na(self, na, target_na, dtype, target_dtype, indexer):
+    def test_setitem_enlarge_with_na(
+        self, na, target_na, dtype, target_dtype, indexer, warn
+    ):
```

Comment:
note how here, the warnings are only ever raised when there is no enlargement  (i.e. when the indexer is `1`, which is within-bounds)